Python/bin/sexVals.py

```diff
@@ -278,9 +280,28 @@ def sexDateToISO(sexDate, intsec=False, mpc_rounding=True):
       elif intsec:
          # do not report microseconds
          digits = 0
-      else:
-         # pass through at microsecond precision
+      elif microsec:
+         # report microseconds
          digits = 6
+      else:
+         # smart rounding for ISO format seconds according to the sexTime day precision
+         # prec_day prec_sec microsecond digits
+         # 1e-06    9e-08     8
+         # 1e-05    9e-07     7
+         # 0.0001   9e-06     6
+         # 0.001    9e-05     5
+         # 0.01     9e-04     4
+         # 0.1      9e-03     3
+         # 1        9e-02     2
+         # 10       9e-01     1
+         # 100      9e+00     0
+         # 1000     9e+01    -1
+         # 10000    9e+02    -2
+         # 100000   9e+03    -3
+         # 1000000  9e+04    -4
+         # ISO format accepts at most 6 digits (microsecond)
+         # and at minimum 0 digits (whole seconds)
+         digits = int(min(max(0, 2 - math.log10(prec)), 6))
```

Comment:
Just wondering if round() is safer than int() here.

Reply:
What is inside the `int` always resolves to a floating point representation of an integer, like 6.0, since `prec` is a whole power of 10. So I think it's safe to just use int.